### PR TITLE
restructure all docs under new sections

### PR DESCRIPTION
**Pull request by tsloughter**
I've tried to structure the docs better and combine like docs under sections together to make discovery easier.

### CODE REVIEW

File: content/en/_index.html [hunk 1]
```diff
@@ -30,12 +30,12 @@
 Write, test, and maintain projects the Erlang way.
 {{% /blocks/feature %}}
 
-{{% blocks/feature icon="fas fa-box" title="Package" url="/docs/publishing-packages/" %}}
+{{% blocks/feature icon="fas fa-box" title="Package" url="/docs/package_management/publishing-packages/" %}}
```
Comment (ferd) [inline content/en/_index.html:33]: this is gonna break hex.pm links and will need to update there too

**Review by ferd — APPROVED** [review]
I'm a bit more nervous about configuration/ and releases/ being moved since we probably have plenty of external links with anchor to these pages, though they also likely all have v3/ in the path there. We should audit adopting erlang

**Comment by tsloughter** [comment]
Yea, I'm nervous too but really think this needs to be done. The current list of docs is too unstructured.

One thing I'm going to do before this is merged is make sure 404s at least go to the main docs page.

**Comment by ferd** [comment]
See https://github.com/hexpm/hexpm/pull/963 for all the URLs that have just been updated on the hex site, we should keep these in sync, along with Adopting Erlang.

**Pull request merged**